back/engines/commercial/id_vienna_saml/app/lib/id_vienna_saml/citizen_saml_omniauth.rb

```diff
@@ -72,5 +76,32 @@ def redirect_uri(configuration)
     def vienna_login_env
       AppConfiguration.instance.settings('vienna_citizen_login', 'environment').to_sym
     end
+
+    # Generates a placeholder name based on the email.
+    # For first names, uses the first character of the email,
+    # For last names, uses the second character of the email or the first character of the second word if available.
```

Comment:
Don't you want to add these conditions to the tests? Also, what if an email has only one character before `@`? (e.g., using a private domain)

Reply:
Yes I can add more tests here that covers all the use cases. Didn't think of an email address with only one character, good spot! :grimacing: